esmvaltool/diag_scripts/iht_toa/single_model_diagnostics.py

```diff
@@ -0,0 +1,714 @@
+"""(C) Crown Copyright 2023, the Met Office.
+
+Single model diagnostics
+1. Solve the Poisson solver
+2. Produce and save plots
+"""
+
+import datetime
+import logging
+from copy import deepcopy
+from pathlib import Path
+
+import cartopy.crs as ccrs
+import iris
+import iris.plot as iplt
+import matplotlib.dates as mdates
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import gridspec, rcParams
+from poisson_solver import SphericalPoisson
+
+from esmvaltool.diag_scripts.shared import (
+    group_metadata,
+    run_diagnostic,
+    save_figure,
+)
+
+# Initialise logger
+logger = logging.getLogger(Path(__file__).stem)
+
+rcParams.update({
+    'font.size': 14,
+    'xtick.major.pad': 10,
+    'ytick.major.pad': 10,
+    'xtick.major.size': 10,
+    'ytick.major.size': 10,
+    'xtick.minor.size': 5,
+    'ytick.minor.size': 5,
+    'axes.linewidth': 2,
+    'lines.markersize': 8,
+    'lines.linewidth': 2
+})
+
+
+def get_provenance_record(plot_type, ancestor_files):
+    """Create a provenance record describing the diagnostic data and plot."""
+    record = {
+        'statistics': ['mean'],
+        'domains': ['global'],
+        'plot_types': [plot_type],
+        'authors': [
+            'andela_bouwe',
+            'righi_mattia',
+        ],
+        'references': [
+            'acknow_project',
```

Comment:
```suggestion
            'pearce23jclim',
```

Reply:
I've simplified the provenance record by deleting that information, already available in the recipe's yml file.

Reply:
Thanks, that's fine. Just as an explanation of why we have the option of specifying it in the diagnostic script too: if someone else would write a different recipe that uses this diagnostic code, the reference would be lost.

Reply:
Thanks for the explanation. In that case it makes sense to keep the reference to the paper. I've done this in aec22df9f6b0ef9aaa2d5e23c0d90cafd19ba69e.